### ISSUE TITLE
Add variable to show configuration values

### ISSUE THREAD

**Issue by AaronRobinsonMSFT** [dotnet/runtime#50400]
I was recently reminded of a JVM flag that is really useful. The flag is `-XX:+PrintFlagsFinal` and it prints out the configuration of the JVM after it has read in all supplied configuration flags and options.

The proposal would be to add a new environment variable to dump these settings out at run time. For example, setting `COMPlus_PRINTCONFIG=1` would dump out all of the `COMPlus_` based values:

```
String values:
                              1234567890 = 123456
          123456789012345678901234567890 = 12345678901234567890
                                     123 = 1234567890123456789012345678901234567890
                                     123 = (null)
Integer values:
                              1234567890 =      123456 (0x0001e240) {default:      123456 (0x0001e240)}
          123456789012345678901234567890 =          -1 (0xffffffff) {default:      123456 (0x0001e240)}
                                     123 =          -1 (0xffffffff) {default:          -1 (0xffffffff)}
```

**Comment by danmoseley** [comment]
Are there any other potential sources of config flags (e.g. json file) that this dump could aggregate as well?

**Comment by AaronRobinsonMSFT** [comment]
>Are there any other potential sources of config flags (e.g. json file) that this dump could aggregate as well?

Absolutely! There is a boat load of options that exist. The [JIT](https://github.com/dotnet/runtime/blob/main/src/coreclr/jit/jitconfigvalues.h) and [GC](https://github.com/dotnet/runtime/blob/main/src/coreclr/gc/gcconfig.h) have many options that are not defined in the [VM](https://github.com/dotnet/runtime/blob/main/src/coreclr/inc/clrconfigvalues.h) list, the configuration knobs contained in `*.runtimeconfig.json` file has options that aren't captured in this list too. The ones found in `*.runtimeconfig.json` are typically all what we would call "external" or publicly supported. However, the VM, JIT, and GC have others that we call "internal". These "internal" options are used for runtime development purposes and subject to change.

I think having a way to dump them all out and get an idea on what the current value is and/or its default would be very helpful and make for a more friendly experience in configuring the runtime.

**Comment by elinor-fung** [comment]
Everything that gets set via `.runtimeconfig.json` (all properties, not just special ones that the runtime reads) should end up as a [configuration knob](https://github.com/dotnet/runtime/blob/main/src/coreclr/utilcode/configuration.cpp). I think the 'external' (runtimeconfig) options generally also map to an 'internal' option (which gets used if the 'external' option isn't set).

**Comment by davidfowl** [comment]
Show both hex and decimal in parentheses

**Comment by AaronRobinsonMSFT** [comment]
I investigated this a bit. The WIP branch is at https://github.com/AaronRobinsonMSFT/runtime/tree/runtime_log_config. I've been able to reduce the Windows Release build size impact to 12k. I am not sure having this functionality is worth it for the 12k impact at present.
